redis/cache/redisx/redisxtest/redisxtest.go

```diff
@@ -0,0 +1,43 @@
+package redisxtest
+
+import (
+	"context"
+	"time"
+
+	"github.com/alicebob/miniredis/v2"
+	"github.com/joomcode/redispipe/redis"
+	"github.com/joomcode/redispipe/redisconn"
+
+	"github.com/reddit/baseplate.go/redis/cache/redisx"
+)
+
+// NewMockRedisClient sets up a mock redis cluster, client and sender
+// This should be called from TestMain since the miniredis instance created is locked
+func NewMockRedisClient(ctx context.Context, timeout time.Duration) (client redisx.Syncx, teardown func(), err error) {
+	redisCluster, err := miniredis.Run()
+	if err != nil {
+		return redisx.Syncx{}, func() {}, err
+	}
+
+	conn, err := redisconn.Connect(ctx, redisCluster.Addr(), redisconn.Opts{IOTimeout: timeout})
+	if err != nil {
+		redisCluster.Close()
+		return redisx.Syncx{}, func() {}, err
+	}
+
+	client = redisx.Syncx{
+		Sync: redisx.BaseSync{SyncCtx: redis.SyncCtx{S: conn}},
+	}
+
+	// Teardown closure
+	teardown = func() {
+		redisCluster.Close()
+		conn.Close()
+	}
+
+	return client, teardown, nil
+}
+
+func FlushRedis(ctx context.Context, client redisx.Syncx) error {
```

Comment:
I don't see this used by `NewMockRedisClient`, and it's not explained by the PR description either.

Reply:
Removing. This was a func we were using locally that got dropped in here. Agree it is probably not necessary.